lib/services/android-project-service.ts

```diff
@@ -396,18 +399,24 @@ export class AndroidProjectService extends projectServiceBaseLib.PlatformProject
 
 			let buildOptions = this.getBuildOptions();
 
-			buildOptions.unshift("clean");
-
 			let projectRoot = this.platformData.projectRoot;
-			let gradleBin = this.useGradleWrapper(projectRoot) ? path.join(projectRoot, "gradlew") : "gradle";
-			if (this.$hostInfo.isWindows) {
-				gradleBin += ".bat";
-			}
-			this.spawn(gradleBin, buildOptions, { stdio: "inherit", cwd: this.platformData.projectRoot }).wait();
+
+			this.cleanProject(projectRoot, buildOptions);
 		}
 		return Future.fromResult();
 	}
 
+	private cleanProject(projectRoot: string, options: string[]): void {
```

Comment:
Yes, but think of the async method in a C# manner:
You can use `await` only in methods marked as `async`.
`Futures` allow such inconsistency - to have `.wait` in a pure sync method, but it's a common practice to wrap your code in a Future, when anywhere in the body you are using `.wait`. Also Promises and `async/await` in TypeScript does not allow having `await` without the method being marked as `async`, so it will be easier to replace the `IFuture` with `Promise` in the method definition.